lua/cfc_adverts/server/sv_adverts.lua

```diff
@@ -0,0 +1,77 @@
+CFCAdverts = CFCAdverts or {}
+
+CFCAdverts._advertCount = 0
+CFCAdverts._curAdvert = 1
+CFCAdverts._advertKeys = {}
+CFCAdverts._timerName = CFCAdverts.HOOK_BASE .. "_SendAdvert"
+
+local function buildNotif( notif, advert )
+    notif:SetTitle( advert.title )
+    notif:SetText( advert.text )
+    notif:SetTextColor( advert.textColor )
+    notif:SetDisplayTime( advert.displayTime )
+    notif:SetPriority( advert.priority )
+    notif:SetCloseable( advert.closeable )
+    notif:SetIgnoreable( advert.ignoreable )
+    notif:SetTimed( advert.timed )
+
+    if advert.type == "Buttons" then
+        for i, button in pairs( advert.buttons ) do
+            notif:AddButton( button.text, button.buttonColor, i )
+        end
+
+        function notif:OnButtonPressed( ply, index, ... )
+            local button = advert.buttons[index]
+
+            button.func( ply, arg )
+
+            if not button.closeOnPress then return end
```

Comment:
Might be wrong, but have you tested this with closeOnPress as false? I feel like the notifs already close on press.

Reply:
Yup, they do.